common/com/twitter/intellij/pants/util/PantsUtil.java

```diff
@@ -763,7 +763,15 @@ public static boolean supportExportDefaultJavaSdk(@NotNull final String pantsExe
     }
 
     final String defaultPlatform = exportResult.getJvmPlatforms().getDefaultPlatform();
-    final String jdkName = String.format("JDK from pants %s", defaultPlatform);
+
+    // if defaultPlatform is java8, then jdkName should be 1.8_from_pants
+    String jdkName = "1.x_from_pants";
+    for (String version : ContainerUtil.newArrayList("6", "7", "8", "9")) {
+      if (defaultPlatform.contains(version)) {
```

Comment:
I could imagine circumstances where a platform name is something like jdk_1.9.0_u66. For example, if a repo is testing with different jdk patch releases. Admittedly, a name like that goes against the convention in the pants repo, but it isn't disallowed.
If this checked against the source level rather than the name, it would not be vulnerable to this. And, it's behavior would align better with the meaning of the platform config.